Snaplen

RichardHabeeb opened the pull request:
Changed pcap file snap length handling. Fixes issues when snort is configured to sample more that 128 bytes.

RichardHabeeb committed:
Changed the snap length default from 128. Added configuration option -n for snap length.

RichardHabeeb committed:
Defensively set pcap packet caplen to always be LTE the snaplen. Additional change to len field (this needs more testing but I believe is was being set incorrectly or the variables were assigned incorrectly).

RichardHabeeb committed:
Bug fix for last two commits.

RichardHabeeb committed:
Updated instructions for snap length.

sflow commented:
Since each individual pcap_pkthdr has .len and .caplen fields,  does it not work if we simply announce the snaplen as 0 (or maybe 65535) in the pcap file header?  That way we can faithfully pass on the original packet-size and caplen from each sFlow packet-sample.

RichardHabeeb commented:
Yes, I upped the snapshot length default to 65535 per [the libpcap documentation here](https://wiki.wireshark.org/Development/LibpcapFileFormat). But I also added a provision to allow adjustments to that field at runtime. Lastly I limited the .caplen field (and thus the amount copied into the pcap) to the snapshot length. 

The last thing I changed might need some more review, line 1273 was occasionally producing incorrect results, this fixed it for my case, but I am unsure about other scenarios.

sflow commented:
I finally checked in a change that sets SA_MAX_PCAP_PKT to 65536 and also sets pcap_file_header.snaplen to the same value.   If you think it really should be run-time configurable then please let me know.
